src/cosim/slave_simulator.cpp

```diff
@@ -227,14 +227,10 @@ class set_variable_cache
                     references_.emplace_back(ref);
                     values_.emplace_back(exposedVariables_.at(ref).lastValue);
                 }
+                const auto index = exposedVariables_.at(ref).arrayIndex;
```

Comment:
Since we are talking performance;
`exposedVariables_.at(ref)` is called *at least* two times in this loop. It might make sense to define it as a local variable before line 224.

Reply:
Thanks, updated accordingly.